app/presenters/advanced_search_result_presenter.rb

```diff
@@ -33,4 +34,42 @@ def show_metadata?
 
     true
   end
+
+  def metadata_bundle
+    return unless show_metadata?
+
+    metadata_to_return = Array.new
+
+    if publication_date.is_a? Hash
```

Comment:
@injms

Reply:
I tried and it complained a lot - I thought that if `publication_date` was `nil`, the check for `publication_date.has_key? :label` would throw an error as that it relied on `publication_date` being a `Hash`. Is that a correct assumption?

I also tried both ways around, with `has_key?` being before and after the `&&` and errors were thrown both ways.

Reply:
Updated.

Reply:
```&&``` will evaluate the first argument and if that's false won't evaluate the second argument because the result will always be false so it should not throw an error.
It does need some extra parentheses to make it work so that might be the cause of the errors